tensorrt_llm/llmapi/llm.py

```diff
@@ -447,6 +447,12 @@ def _prepare_sampling_params(
                         "tokenizer is required to reset end_id if it is None, or you can explicitly specify the end_id for sampling_params"
                     )
                 sampling_params._setup(self.tokenizer)
+            if sampling_params.prompt_logprobs:
+                # Context logits are required to compute prompt_logprobs
+                sampling_params.return_context_logits = True
+                # TODO: need a way to know whether to drop context logits at worker later
```

Comment:
@Superjomn  - for me to dump contexts (or generation logits) before sending it via IPC, I need a flag to know whether users actually set `return_context_logits`  or it was enabled here.  

It could be achieved by adding an additional internal flag in `SamplingParams`, lmk if you suggest against that.

Reply:
Yeah, that makes sense. How about introducing it in SamplingParams as a private property? As SamplingParams is a 1.0 committed API and currently, this implementation is just a WAR.
```python
@property
def _need_return_context_logits(self) -> bool:
    return self.prompt_logprobs
```
This shouldn't break the API stability check.